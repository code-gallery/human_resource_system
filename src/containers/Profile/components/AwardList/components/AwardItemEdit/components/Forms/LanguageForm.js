import React from 'react'
import PropTypes from 'prop-types'
import Select from 'react-select'
import { renderValidation } from 'containers/Profile/utils/validation'
import languages from 'utils/languages'
import proficiencyOptions from 'utils/languagesProficiency'
import BaseForm from './BaseForm'

class LanguageForm extends BaseForm {
  componentDidMount() {
    if (!this.props.proficiency) {
      this.handleChange(5, 'proficiency')
    }
  }

  render() {
    const {
      description,
      name,
      proficiency,
      invalidFields
    } = this.props

    return (
      <div>
        <div className="form-group">
          <label className={renderValidation(invalidFields, 'name')}>Name</label>
          <Select
            name="select-language"
            placeholder={name}
            options={languages}
            ignoreCase={true}
            matchProp="label"
            matchPos="any"
            value={name}
            onChange={(obj) => {
              this.onTextChange('name', {
                target: {
                  value: obj.label
                }
              })
            }}
          />
        </div>
        <div className="form-group">
          <label className={renderValidation(invalidFields, 'description')}>Description</label>
          <input
            type="text"
            name="description"
            defaultValue={description}
            onChange={v => this.onTextChange('description', v)}
          />
        </div>
        <div className="form-group">
          <label className={renderValidation(invalidFields, 'proficiency')}>Skill Level</label>
          <Select
            name="select-proficiency"
            options={proficiencyOptions}
            ignoreCase={true}
            matchProp="label"
            matchPos="any"
            value={parseInt(proficiency, 10)}
            onChange={(obj) => {
              if (obj && obj.value) {
                this.handleChange(obj.value, 'proficiency')
              }
            }}
          />
        </div>
      </div>
    )
  }
}

LanguageForm.defaultProps = {
  invalidFields: []
}

LanguageForm.propTypes = {
  invalidFields: PropTypes.array,
  name: PropTypes.string,
  description: PropTypes.string,
  proficiency: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string
  ]),
  onFieldChange: PropTypes.func.isRequired
}

export default LanguageForm
