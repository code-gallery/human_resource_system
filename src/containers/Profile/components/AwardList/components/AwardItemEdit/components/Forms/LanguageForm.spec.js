import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import LanguageForm from './LanguageForm'

describe('containers/Profile/components/.../LanguageForm', () => {
  const commonProps = {
    onFieldChange: jest.fn()
  }

  describe('when all inputs have no values', () => {
    it('renders correctly', () => {
      const props = {
        ...commonProps
      }
      const tree = renderer.create(
        <LanguageForm {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when all inputs have values', () => {
    const props = {
      ...commonProps,
      proficiency: 5,
      name: 'English',
      description: 'Lorem ipsum'
    }

    it('renders correctly', () => {
      const tree = renderer.create(
        <LanguageForm {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })

    describe('componentDidMount', () => {
      it('calls props.onFieldChange correctly', () => {
        shallow(
          <LanguageForm {...props} />
        )
        expect(props.onFieldChange)
          .toHaveBeenCalledWith('proficiency', 5)
      })
    })

    describe('handleChange', () => {
      describe('when "description" changes', () => {
        it('calls props.onFieldChange correctly', () => {
          const wrapper = shallow(
            <LanguageForm {...props} />
          )
          const event = { target: { name: 'description', value: 'hello world' } }
          props.onFieldChange.mockClear()
          wrapper.find('input[name="description"]').simulate('change', event)
          expect(props.onFieldChange)
            .toHaveBeenCalledWith('description', 'hello world')
        })
      })
    })
  })
})
