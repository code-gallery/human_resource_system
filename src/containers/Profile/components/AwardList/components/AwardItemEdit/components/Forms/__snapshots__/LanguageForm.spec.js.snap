// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Profile/components/.../LanguageForm when all inputs have no values renders correctly 1`] = `
<div>
  <div
    className="form-group"
  >
    <label
      className=""
    >
      Name
    </label>
    <div
      className="Select is-clearable is-searchable Select--single"
      style={undefined}
    >
      <div
        className="Select-control"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <span
          className="Select-multi-value-wrapper"
          id="react-select-2--value"
        >
          <div
            className="Select-placeholder"
          >
            Select...
          </div>
          <div
            className="Select-input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-activedescendant="react-select-2--value"
              aria-describedby={undefined}
              aria-expanded="false"
              aria-haspopup="false"
              aria-label={undefined}
              aria-labelledby={undefined}
              aria-owns=""
              className={undefined}
              id={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              role="combobox"
              style={
                Object {
                  "boxSizing": "content-box",
                  "width": "5px",
                }
              }
              tabIndex={undefined}
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </span>
        <span
          className="Select-arrow-zone"
          onMouseDown={[Function]}
        >
          <span
            className="Select-arrow"
            onMouseDown={[Function]}
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
    >
      Description
    </label>
    <input
      defaultValue={undefined}
      name="description"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
    >
      Skill Level
    </label>
    <div
      className="Select is-clearable is-searchable Select--single"
      style={undefined}
    >
      <div
        className="Select-control"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <span
          className="Select-multi-value-wrapper"
          id="react-select-3--value"
        >
          <div
            className="Select-placeholder"
          >
            Select...
          </div>
          <div
            className="Select-input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-activedescendant="react-select-3--value"
              aria-describedby={undefined}
              aria-expanded="false"
              aria-haspopup="false"
              aria-label={undefined}
              aria-labelledby={undefined}
              aria-owns=""
              className={undefined}
              id={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              role="combobox"
              style={
                Object {
                  "boxSizing": "content-box",
                  "width": "5px",
                }
              }
              tabIndex={undefined}
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </span>
        <span
          className="Select-arrow-zone"
          onMouseDown={[Function]}
        >
          <span
            className="Select-arrow"
            onMouseDown={[Function]}
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`containers/Profile/components/.../LanguageForm when all inputs have values renders correctly 1`] = `
<div>
  <div
    className="form-group"
  >
    <label
      className=""
    >
      Name
    </label>
    <div
      className="Select is-clearable is-searchable Select--single"
      style={undefined}
    >
      <div
        className="Select-control"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <span
          className="Select-multi-value-wrapper"
          id="react-select-4--value"
        >
          <div
            className="Select-placeholder"
          >
            English
          </div>
          <div
            className="Select-input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-activedescendant="react-select-4--value"
              aria-describedby={undefined}
              aria-expanded="false"
              aria-haspopup="false"
              aria-label={undefined}
              aria-labelledby={undefined}
              aria-owns=""
              className={undefined}
              id={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              role="combobox"
              style={
                Object {
                  "boxSizing": "content-box",
                  "width": "5px",
                }
              }
              tabIndex={undefined}
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </span>
        <span
          className="Select-arrow-zone"
          onMouseDown={[Function]}
        >
          <span
            className="Select-arrow"
            onMouseDown={[Function]}
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
    >
      Description
    </label>
    <input
      defaultValue="Lorem ipsum"
      name="description"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
    >
      Skill Level
    </label>
    <div
      className="Select has-value is-clearable is-searchable Select--single"
      style={undefined}
    >
      <input
        disabled={false}
        name="select-proficiency"
        type="hidden"
        value="5"
      />
      <div
        className="Select-control"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <span
          className="Select-multi-value-wrapper"
          id="react-select-5--value"
        >
          <div
            className="Select-value"
            style={undefined}
            title={undefined}
          >
            <span
              aria-selected="true"
              className="Select-value-label"
              id="react-select-5--value-item"
              role="option"
            >
              Intermediate - Mid (A2)
            </span>
          </div>
          <div
            className="Select-input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-activedescendant="react-select-5--value"
              aria-describedby={undefined}
              aria-expanded="false"
              aria-haspopup="false"
              aria-label={undefined}
              aria-labelledby={undefined}
              aria-owns=""
              className={undefined}
              id={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              role="combobox"
              style={
                Object {
                  "boxSizing": "content-box",
                  "width": "5px",
                }
              }
              tabIndex={undefined}
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </span>
        <span
          aria-label="Clear value"
          className="Select-clear-zone"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          title="Clear value"
        >
          <span
            className="Select-clear"
            dangerouslySetInnerHTML={
              Object {
                "__html": "&times;",
              }
            }
          />
        </span>
        <span
          className="Select-arrow-zone"
          onMouseDown={[Function]}
        >
          <span
            className="Select-arrow"
            onMouseDown={[Function]}
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
